File: src/main/kotlin/paradigms/concureent/IntroToThreads.java
package paradigms.concureent;

public class IntroToThreads {


    public static void startThreadPooling() {
        for (int i = 0; i < 2; i++) {
            int finalT = i;
            new Thread(() -> {
                for (int j = 0; j < 100; j++) {
                    System.out.println("Thread" + finalT + ": " + j);
                }
            }
            ).start();

        }
    }

    public static void main(String[] args) {


        startThreadPooling();
    }
}

File: src/main/kotlin/paradigms/concureent/IntroToThreads.kt
package org.example.paradigms.concureent

import paradigms.concureent.IntroToThreads
import kotlin.concurrent.thread


private fun startThreadPooling() {

    repeat(2) {
        thread {
            for (i in 1..100) {
                println("Thread:${it} : ${i}")
            }
        }
    }
}

fun main() {
    val manualThread = thread(start = false) {
        println("I will not start until explicitly started")
    }
    val autoThread = thread {
        println("I am auto starting ")
    }

    manualThread.start()
}